chia/wallet/wallet_transaction_store.py

```diff
@@ -65,22 +65,20 @@ async def create(cls, db_wrapper: DBWrapper2):
             await conn.execute(
                 "CREATE INDEX IF NOT EXISTS tx_confirmed_index on transaction_record(confirmed_at_height)"
             )
-
             await conn.execute("CREATE INDEX IF NOT EXISTS tx_created_index on transaction_record(created_at_time)")
-
+            # Remove a redundant index on `created_at_time`
+            # See https://github.com/Chia-Network/chia-blockchain/issues/10276
+            await conn.execute("DROP INDEX IF EXISTS tx_created_time")
+            await conn.execute("CREATE INDEX IF NOT EXISTS tx_to_puzzle_hash on transaction_record(to_puzzle_hash)")
             await conn.execute("CREATE INDEX IF NOT EXISTS tx_confirmed on transaction_record(confirmed)")
-
             await conn.execute("CREATE INDEX IF NOT EXISTS tx_sent on transaction_record(sent)")
-
-            await conn.execute("CREATE INDEX IF NOT EXISTS tx_created_time on transaction_record(created_at_time)")
-
-            await conn.execute("CREATE INDEX IF NOT EXISTS tx_type on transaction_record(type)")
-
-            await conn.execute("CREATE INDEX IF NOT EXISTS tx_to_puzzle_hash on transaction_record(to_puzzle_hash)")
-
             await conn.execute(
                 "CREATE INDEX IF NOT EXISTS transaction_record_wallet_id on transaction_record(wallet_id)"
             )
+            await conn.execute(
+                "CREATE INDEX IF NOT EXISTS transaction_record_trade_id_idx ON transaction_record(trade_id)"
```

Comment:
what's the reason to add this index? It's not mentioned in neurosis' ticket, as far as I can tell

Reply:
I mentioned "inspired" in the description, because while checking those, I found `get_transactions_by_trade_id` and it's doing `(2, 0, 0, 'SCAN TABLE transaction_record')` which is.. not good.

With the index it does `(3, 0, 0, 'SEARCH TABLE transaction_record USING INDEX transaction_record_trade_id_idx (trade_id=?)')` as it should.